### ISSUE TITLE
almost_swapped with let

### ISSUE THREAD

**Issue by xNxExOx** [rust-lang/rust-clippy#8151]
### Summary

```rs
a = b;
b = a;
```
produces lint correctly, but adding `let` does not:
```rs
let a = b;
let b = a;
```

### Lint Name

almost_swapped

### Reproducer

I tried this code:

```rust
let mut a = 1;
let mut b = 2;
let a = b;
let b = a;
```

I expected to see this happen:

Instead, this happened:
```
error: this looks like you are trying to swap `a` and `b`
   --> src\patches\inventory\filter.rs:255:5
    |
3  | /     let a = b;
4  | |     let b = a;
    | |_________^ help: try: `std::mem::swap(&mut a, &mut b)`
    |
    = note: `#[deny(clippy::almost_swapped)]` on by default
    = note: or maybe you should use `std::mem::replace`?
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_swapped

```

### Version

```text
rustc 1.59.0-nightly (efec54529 2021-12-04)
binary: rustc
commit-hash: efec545293b9263be9edfb283a7aa66350b3acbf
commit-date: 2021-12-04
host: x86_64-pc-windows-msvc
release: 1.59.0-nightly
LLVM version: 13.0.0
```
